server/manager/src/main/java/org/apache/accumulo/manager/compaction/queue/ResolvedCompactionJob.java

```diff
@@ -0,0 +1,207 @@
+import org.apache.accumulo.core.spi.compaction.CompactorGroupId;
+
+/**
+ * This class takes a compaction job and the tablet metadata from which is was generated and
+ * computes all information that is needed to actually run the compaction. A goal of this class is
+ * to avoid referencing anything that is not needed. For example only a subset of the information
+ * from TabletMetadata is needed, therefore a reference to the TabletMetadata object should not be
+ * kept. This object will placed on a queue for compaction and that is why it should avoid using
+ * more memory than needed. Before this class was created queued compaction jobs would reference a
+ * tablet metadata object which included all of the tablets files. When tablets had lots of file and
+ * lots compactions against them this could use lots of memory and make the manager unstable at time
+ * when things were already bad.
+ */
+public class ResolvedCompactionJob implements CompactionJob {
+
+  // The fate id of the table compaction that selected files
+  private final FateId selectedFateId;
+  private final Map<StoredTabletFile,DataFileValue> jobFiles;
+  private final CompactionKind kind;
+  private final boolean compactingAll;
+  private final KeyExtent extent;
+  private final short priority;
+  private final CompactorGroupId group;
+  private final String tabletDir;
+  private final boolean overlapsSelectedFiles;
+
+  private static long weigh(Range range) {
+    long estDataSize = 0;
+    if (range != null) {
+      var staryKey = range.getStartKey();
+      estDataSize += staryKey == null ? 0 : staryKey.getSize();
+      var endKey = range.getEndKey();
+      estDataSize += endKey == null ? 0 : endKey.getSize();
+    }
+    return estDataSize;
+  }
+
+  public static final SizeTrackingTreeMap.Weigher<CompactionJob> WEIGHER = job -> {
+    if (job instanceof ResolvedCompactionJob) {
+      var rcj = (ResolvedCompactionJob) job;
+      long estDataSize = 0;
+      if (rcj.selectedFateId != null) {
+        estDataSize += rcj.selectedFateId.canonical().length();
+      }
+      for (var file : rcj.jobFiles.keySet()) {
+        estDataSize += file.getMetadataPath().length();
+        estDataSize += 24; // There are three longs in DataFileValue
+        estDataSize += weigh(file.getRange());
+      }
+
+      estDataSize += rcj.group.canonical().length();
+      estDataSize += rcj.extent.tableId().canonical().length();
+      estDataSize += rcj.extent.prevEndRow() == null ? 0 : rcj.extent.prevEndRow().getLength();
+      estDataSize += rcj.extent.endRow() == null ? 0 : rcj.extent.endRow().getLength();
+      estDataSize += rcj.tabletDir.length();
+      // Guess at how many bytes the overhead at things like pointers and primitives are taking in
+      // this object.
+      estDataSize += 64;
+      return estDataSize;
+    } else {
+      // Do not know the concrete type so weigh based on the interface methods of CompaactionJob
```

Comment:
```suggestion
      // Do not know the concrete type so weigh based on the interface methods of CompactionJob
```

Reply:
I missed this comment when merging.  Pushed aa1f30aa64fd45ff0844fe1e13b749c85643472e to main to pull in this suggestion.